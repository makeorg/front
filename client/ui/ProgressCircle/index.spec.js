import React from 'react';
import renderer from 'react-test-renderer';
import { ProgressCircleComponent } from '.';

describe('ProgressCircleComponent', () => {
  it('toMatchSnapshot', () => {
    const component = renderer
      .create(
        <ProgressCircleComponent index={2} cardsCount={12} cardOffset={1} />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
