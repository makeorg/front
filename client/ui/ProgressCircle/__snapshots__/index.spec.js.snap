// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressCircleComponent toMatchSnapshot 1`] = `
Array [
  .c0 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

@media (min-width:767px) {

}

@media (min-width:360px) {

}

@media (min-width:969px) {

}

@media (min-width:767px) {

}

<progress
    aria-valuemax={13}
    aria-valuenow={3}
    className="c0"
    data-cy-element="progress-a11y"
    max={13}
    value={3}
  >
    proposal_card.number
  </progress>,
  .c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 58px;
  height: 58px;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 0;
  width: 100%;
  height: 100%;
}

.c2 {
  fill: rgba(255,255,255,1);
}

.c3 {
  fill: transparent;
  stroke: rgba(0,0,0,0.05);
  stroke-width: 3px;
}

.c4 {
  fill: transparent;
  stroke-width: 3px;
  stroke-dasharray: 23 77;
  stroke-dashoffset: 25;
}

.c5 {
  font-size: 12px;
  line-height: 18px;
  color: rgba(0,0,0,0.65);
  position: relative;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  z-index: 1;
  font-family: 'Circular Standard Book',Arial,sans-serif;
}

.c6 {
  font-size: 12px;
  color: rgba(0,0,0,1);
}

@media (min-width:767px) {
  .c0 {
    width: 82px;
    height: 82px;
  }
}

@media (min-width:360px) {
  .c5 {
    font-size: 14px;
    line-height: 21px;
  }
}

@media (min-width:969px) {
  .c5 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:767px) {
  .c6 {
    font-size: 18px;
  }
}

<div
    aria-hidden={true}
    className="c0"
    data-cy-container="progress"
  >
    <svg
      className="c1"
      height={42}
      viewBox="0 0 42 42"
      width={42}
    >
      <circle
        className="c2"
        cx={21}
        cy={21}
        r={15.91549430918954}
      />
      <circle
        className="c3"
        cx={21}
        cy={21}
        r={15.91549430918954}
      />
      <circle
        className="c4"
        cx={21}
        cy={21}
        r={15.91549430918954}
      />
    </svg>
    <p
      className="c5"
    >
      <span
        className="c6"
      >
        3
      </span>
      /13
    </p>
  </div>,
]
`;
